source/agora/node/Registry.d

```diff
@@ -897,7 +929,7 @@ private struct ZoneData
                 // fields of data from a primary, it redirects API calls to API of the
                 // configured primary
                 this.redirect_register = this.netman.makeRegistryClient(
-                    this.config.redirect_register);
+                    Address(this.config.redirect_register));
```

Comment:
What about changing the field type in Config?

Reply:
Will probably do it, but I think we're still figuring out the right combination of fields, so I just delayed it.
Ideally I'd like to reduce the number of different ways we configure upstream registries down to 0, with overrides available.